### DISCUSSION TITLE
FE launch success, but cannot visit

### DISCUSSION THREAD

**Discussion by jacentsao** [apache/doris#10964] [A - General / Q&A] [answered]
* below is fe.log

```
[Catalog.waitForReady():848] wait catalog to be ready. FE type: MASTER. is ready: true
[Catalog.waitForReady():843] catalog is ready. FE type: MASTER
[Xnio.<clinit>():95] XNIO version 3.6.5.Final2022-07-18 00:26:33,378 INFO (UNKNOWN 192.168.52.128_9010_1658110932495(-1)|1) [NioXnio.<clinit>():59] XNIO NIO Implementation Version 3.6.5.Final
[FeServer.start():48] thrift server started.
[RestartApplicationListener.onApplicationStartingEvent():93] Restart disabled due to System property 'spring.devtools.restart.enabled' being set to false
] HV000001: Hibernate Validator 5.1.0.Final
[StartupInfoLogger.logStarting():55] Starting PaloFe v1.0-SNAPSHOT using Java 11.0.15 on localhost.localdomain with PID 13568 (/opt/doris/fe/lib/doris-fe.jar started by root in /opt/doris)
[SpringApplication.logStartupProfileInfo():640] No active profile set, falling back to 1 default profile: "default"
[DeferredLog.logTo():255] For additional web related logging consider setting the 'logging.level.web' property to 'DEBUG'
[RepositoryConfigurationDelegate.registerRepositoriesIn():132] Bootstrapping Spring Data LDAP repositories in DEFAULT mode.
[RepositoryConfigurationDelegate.registerRepositoriesIn():201] Finished Spring Data repository scanning in 6 ms. Found 0 LDAP repository interfaces.
[JettyServletWebServerFactory.getWebServer():166] Server initialized with port: 8030
[ServletWebServerApplicationContext.prepareWebApplicationContext():290] Root WebApplicationContext: initialization completed in 977 ms
[FrameworkServlet.initServletBean():525] Initializing Servlet 'dispatcherServlet'
[FrameworkServlet.initServletBean():547] Completed initialization in 1 ms
[JettyWebServer.start():172] Jetty started on port(s) 8030 (http/1.1) with context path '/'
[StartupInfoLogger.logStarted():61] Started PaloFe in 1.82 seconds (JVM running for 5.586)
[NMysqlServer.start():73] Open mysql server success on 9030
[QeService.start():65] QE service start.
```
* after fe launch success, visit default 8030 port returns 404

![image](https://user-images.githubusercontent.com/16348792/179446034-fea8e583-4940-4b05-b6d8-702e2826fbe1.png)

* is there any additional config  to fix this problem? many thanks

**Comment by hf200012** [comment]
在fe.out文件里，能看到这个信息才说明真正的启动起来
![image](https://user-images.githubusercontent.com/820301/179446924-e686a296-8c98-4338-b031-e82c2debd5af.png)

**Reply by jacentsao** [reply]
有看到这些日志，但是仍然无法访问
![image](https://user-images.githubusercontent.com/16348792/179452079-5ec39838-dc2d-4069-abbd-c9babe984ccc.png)

**Comment by sun-help** [comment, marked as answer]
> 

需要发一下完整日志

**Reply by jacentsao** [reply]
> > 
> 
> 需要发一下完整日志

感谢回复，日志如文件所示

[log.zip](https://github.com/apache/doris/files/9129631/log.zip)

**Reply by sun-help** [reply]
> 

你好
1.你的doris版本是多少
2.您下载的分支是哪一个
3.您用什么编译的

**Reply by jacentsao** [reply]
你好
1&2  使用的7.16号的master分支
```
commit 75ca21dafaeb197f31990a28e427135f6a9eb2c5 (grafted, HEAD -> master, origin/master, origin/HEAD)
Author: Gabriel <gabrielleebuaa@gmail.com>
Date:   Sat Jul 16 14:18:38 2022 +0800

    [Bug] handle null map right in vectorized load (#10883)
```
3.  使用的是https://github.com/apache/doris/tree/master/docker   这个教程，Windows下面docker编译，centos7部署

我重新用官方编译好的fe可以正常访问，不知道是哪一步出问题了，感谢回复

**Reply by sun-help** [reply]
> 你好 1&2 使用的7.16号的master分支
> 
> ```
> commit 75ca21dafaeb197f31990a28e427135f6a9eb2c5 (grafted, HEAD -> master, origin/master, origin/HEAD)
> Author: Gabriel <gabrielleebuaa@gmail.com>
> Date:   Sat Jul 16 14:18:38 2022 +0800
> 
>     [Bug] handle null map right in vectorized load (#10883)
> ```
> 
> 3. 使用的是https://github.com/apache/doris/tree/master/docker   这个教程，Windows下面docker编译，centos7部署
> 
> 我重新用官方编译好的fe可以正常访问，不知道是哪一步出问题了，感谢回复

你是不没有加 --ui ，在编译fe的时候。另外有一个小tips，不要用master的分支，用稳定的分支。

**Reply by sun-help** [reply]
> 你好 1&2 使用的7.16号的master分支
> 
> ```
> commit 75ca21dafaeb197f31990a28e427135f6a9eb2c5 (grafted, HEAD -> master, origin/master, origin/HEAD)
> Author: Gabriel <gabrielleebuaa@gmail.com>
> Date:   Sat Jul 16 14:18:38 2022 +0800
> 
>     [Bug] handle null map right in vectorized load (#10883)
> ```
> 
> 3. 使用的是https://github.com/apache/doris/tree/master/docker   这个教程，Windows下面docker编译，centos7部署
> 
> 我重新用官方编译好的fe可以正常访问，不知道是哪一步出问题了，感谢回复

如果你是第一次使用 build-env-for-0.15.0 或之后的版本，第一次编译的时候要使用如下命令：

sh build.sh --clean --be --fe --ui

这是因为 build-env-for-0.15.0 版本镜像升级了 thrift(0.9 -> 0.13)，需要通过 --clean 命令强制使用新版本的 thrift 生成代码文件，否则会出现不兼容的代码。
！！！不要用master的代码哟

**Reply by jacentsao** [reply]
> > 你好 1&2 使用的7.16号的master分支
> > ```
> > commit 75ca21dafaeb197f31990a28e427135f6a9eb2c5 (grafted, HEAD -> master, origin/master, origin/HEAD)
> > Author: Gabriel <gabrielleebuaa@gmail.com>
> > Date:   Sat Jul 16 14:18:38 2022 +0800
> > 
> >     [Bug] handle null map right in vectorized load (#10883)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 3. 使用的是https://github.com/apache/doris/tree/master/docker   这个教程，Windows下面docker编译，centos7部署
> > 
> > 我重新用官方编译好的fe可以正常访问，不知道是哪一步出问题了，感谢回复
> 
> 如果你是第一次使用 build-env-for-0.15.0 或之后的版本，第一次编译的时候要使用如下命令：
> 
> sh build.sh --clean --be --fe --ui
> 
> 这是因为 build-env-for-0.15.0 版本镜像升级了 thrift(0.9 -> 0.13)，需要通过 --clean 命令强制使用新版本的 thrift 生成代码文件，否则会出现不兼容的代码。 ！！！不要用master的代码哟

好的，非常感谢！！！
